src/backend/api/meals.js

```diff
@@ -1,14 +1,203 @@
+      meals = query.where('when', '>', dateAfter);
+    }
+  }
+
+  if ('dateBefore' in req.query) {
+    const dateBefore = new Date(req.query.dateAfter);
+
+    if (!dateBefore) {
+      res.status(404).json({ error: 'You did not enter date' });
+    } else {
+      meals = query.where('when', '<', dateBefore);
+    }
+  }
+
+  if ('limit' in req.query) {
+    const limit = parseInt(req.query.limit);
+
+    if (!limit) {
+      res.status(404).json({ error: 'You did not enter limit' });
+    } else {
+      meals = query.limit(limit);
+    }
+  }
+
+  if ('sortKey' in req.query) {
+    const sortKey = String(req.query.sortKey);
+
+    if (!sortKey) {
+      res.status(404).json({ error: 'You did not enter sort key' });
+    } else if (sortKey === 'price') {
+      meals = query.orderByRaw(sortKey);
+    } else if (sortKey === 'when') {
+      meals = query.orderBy(sortKey);
+    } else if (sortKey === 'max_reservations') {
+      meals = query.orderByRaw(sortKey);
+    }
+  }
+
+  if ('sortDir' in req.query && 'sortKey' in req.query) {
+    const sortKey = String(req.query.sortKey);
+    const sortDir = String(req.query.sortDir);
+
+    if (!sortKey || !sortDir) {
+      res
+        .status(404)
+        .json({ error: 'You did not enter sort key or direction' });
+    } else if (sortKey === 'price' && sortDir === 'desc') {
+      meals = query.orderBy(sortKey, 'desc');
+    } else if (sortKey === 'when' && sortDir === 'desc') {
+      meals = query.orderBy(sortKey, 'desc');
+    } else if (sortKey === 'max_reservations' && sortDir === 'desc') {
+      meals = query.orderByRaw(sortKey, 'desc');
+    }
+  }
+
+  try {
+    const meal = await query;
+
+    if (meal.length !== 0) {
+      res.send(meal);
+    } else {
+      res.status(200).json({ error: 'No meals found' });
+    }
+  } catch (error) {
+    res.status(500).json({ error: 'Server error' });
+  }
+});
+
+router.post('/', async (req, res) => {
+  try {
+    const newMeal = await knex.insert(req.body).into('meal');
+
+    res.send({ massage: `Meal with ID ${newMeal} created` });
+  } catch (error) {
+    res.status(500).json({ error: 'Server error' });
+  }
+});
+
+router.get('/:id', async (req, res) => {
+  try {
+    const mealId = req.params.id;
+    const meal = await knex.select().from('meal').where({ id: mealId });
+
+    if (meal.length !== 0) {
+      res.send(meal);
+    } else {
+      res.status(404).json({ error: 'Meal not found' });
+    }
+  } catch (error) {
+    res.status(500).json({ error: 'Server error' });
+  }
+});
+
+router.put('/:id', async (req, res) => {
+  try {
+    const mealBody = req.body;
+    const mealId = req.params.id;
+    const meal = await knex('meal').where({ id: mealId });
+
+    if (meal.length === 0) {
+      res.status(400).json({ error: 'Meal not found' });
+    }
+    if (mealBody.length !== 0) {
+      await knex('meal').where({ id: mealId }).update(mealBody);
+      const updatedMeal = await knex('meal').where({ id: mealId });
```

Comment:
You could use single line code for the purpose, `const updatedMeal = await knex('meal').where({ id: mealId }).update(mealBody);`